### ISSUE TITLE
[Bug]: Camera and keyboard stop working in Genshin Impact

### ISSUE THREAD

**Issue by Index-233** [PlayCover/PlayCover#396]
### Describe the bug

In some scenarios, when the CPU occupancy is high, the camera will suddenly fail to work. Dragging the mouse is useless. Clicking the mouse and tapping the keyboard does not respond, and the whole keyboard mapping is frozen. If you press a key (such as the arrow key W) before the keyboard mapping gets stuck, the role will go straight forward and cannot be interrupted. The freeze can only be automatically restored after a few seconds to return to normal. This issue is prevalent and occurs in the stable (1.1.1) build.

### Steps to reproduce

Enter complex scenarios or open more applications to improve CPU usage.

### Expected behaviour

Camera/Mouse panning should be functioning just as fine as the other keys.

### What version of PlayCover are you using?

1.1.1

### What version of macOS are you using?

Monterey (macOS 12)

### Issue Language

- [X] Yes my issue is written in English

**Comment by XuYicong** [comment]
Please try a recent nightly build to see if the issue had been solved.

**Comment by Index-233** [comment]
> Please try a recent nightly build to see if the issue had been solved.

I just tried it.The problem still exists.

**Comment by XuYicong** [comment]
Is it only camera not functioning, or does keyboard also stop working? Your title refers to only camera but your description mentions keyboard. Please rephrase or change at least one of them to make title and description match.

**Comment by amirsaam** [comment]
Unfortunately I think this kind of irresponsible keyboard and mouse in games are an issue in iOS games running on Apple Silicon Mac.

If you have tried iOS games from Mac App Store you'll experience such thing.

In fact it happened to me many times when I play GWENT on my Mac that I've downloaded from Mac App Store.

By the way, have you tried to toggle mouse with option key and typing something in the in-game chat then toggling back and test if keymap work out not?

**Comment by photosunthesis** [comment]
I am experiencing a similar issue with my M2 MacBook Pro. The controls would freeze and the last keys I pressed on my keyboard and mouse remain pressed for a few seconds. Even the `alt` key would freeze from time to time; toggling the cursor would take a few seconds, or sometimes it would not toggle at all. I have tried doing a clean install of everything (PlayCover and Genshin Impact) but the issue still remains, and I am using the latest nightly build as of writing this comment. I also found that only when I use it with simulated touch will the issue occur. When I use my mouse to touch the controls on screen, it does not happen.
